webapp/src/modules/estates/utils.js

```diff
@@ -0,0 +1,17 @@
+export const calculateZoomAndCenter = parcels => {
+  const xs = [...new Set(parcels.map(coords => coords.x).sort())]
+  const ys = [...new Set(parcels.map(coords => coords.y).sort())]
+  const x = xs[parseInt(xs.length / 2, 10)]
+  const y = ys[parseInt(ys.length / 2, 10)]
+  const center = { x, y }
+  const zoom = 1 / (xs.length + ys.length) * 7.5
```

Comment:
Magic number :P ? I haven't tested this branch yet, I'll check it out later, but does this behave well for any estate size? I think we should be able compute the `size` prop instead of the `zoom`, we can do that easily if we have the `width`, `height` and `parcels`, which we should have from any `EstatePreview`.  Most of the code around the map is my fault :P so I'll help you out, no worries.